src/dotnet/install.sh

```diff
@@ -4,465 +4,94 @@
-###########################
-
-export DEBIAN_FRONTEND=noninteractive
-
-. /etc/os-release
-architecture="$(dpkg --print-architecture)"
-
-# Dotnet 3.1 and 5 are not supported on Ubuntu 22.04 (jammy)+,
-# due to lack of libssl3.0 support.
-# See: https://github.com/microsoft/vscode-dev-containers/issues/1458#issuecomment-1135077775
-# NOTE: This will only guard against installation of the dotnet versions we propose via 'features'. 
-#       The user can attempt to install any other version at their own risk.
-if [[ "${DOTNET_VERSION}" = "3"* ]] || [[ "${DOTNET_VERSION}" = "5"* ]]; then
-    if [[ ! "${DOTNET_VERSION_CODENAMES_REQUIRE_OLDER_LIBSSL_1}" = *"${VERSION_CODENAME}"* ]]; then
-        err "Dotnet ${DOTNET_VERSION} is not supported on Ubuntu ${VERSION_CODENAME} due to a change in the 'libssl' dependency across distributions.\n Please upgrade your version of dotnet, or downgrade your OS version."
-        exit 1
-    fi
-fi
-
-# Determine if the user wants to download .NET Runtime only, or .NET SDK & Runtime
-# and set the appropriate variables.
-if [ "${DOTNET_RUNTIME_ONLY}" = "true" ]; then
-    DOTNET_SDK_OR_RUNTIME="runtime"
-elif [ "${DOTNET_RUNTIME_ONLY}" = "false" ]; then
-    DOTNET_SDK_OR_RUNTIME="sdk"
-else
-    err "Expected true for installing dotnet Runtime only or false for installing SDK and Runtime. Received ${DOTNET_RUNTIME_ONLY}."
+if [ "$(id -u)" -ne 0 ]; then
+    echo -e 'Script must be run as root. Use sudo, su, or add "USER root" to your Dockerfile before running this script.'
     exit 1
 fi
 
-# Install the .NET CLI
-echo "(*) Installing .NET CLI..."
+# icu-devtools includes dependencies for .NET
+check_packages wget ca-certificates icu-devtools
 
-CHANGE_OWNERSHIP="false"
-if [[ "${DOTNET_ARCHIVE_ARCHITECTURES}" = *"${architecture}"* ]] && [[  "${DOTNET_ARCHIVE_VERSION_CODENAMES}" = *"${VERSION_CODENAME}"* ]] && [[ "${INSTALL_USING_APT}" = "true" ]]; then
-    echo "Detected ${VERSION_CODENAME} on ${architecture}. Attempting to install dotnet from apt"
+wget -O "$DOTNET_INSTALL_SCRIPT" "$DOTNET_INSTALL_SCRIPT_URL"
+chmod +x "$DOTNET_INSTALL_SCRIPT"
 
-    install_using_apt "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}" false || install_using_apt "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}" true
-    if [ "$?" != 0 ]; then
-        echo "Could not install requested version from apt on current distribution."
-        exit 1
-    fi
-else
-    if [[ "${INSTALL_USING_APT}" = "false" ]]; then
-        echo "Installing dotnet from releases url"
-    else
-        echo "Attempting to install dotnet from releases url"
-    fi
-    install_using_dotnet_releases_url "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_VERSION}"
-    CHANGE_OWNERSHIP="true"
-fi
+# Install primary version
+install_version "$DOTNET_VERSION"
 
-# Additional dotnet versions to be installed but not be set as default.
-if [ ! -z "${ADDITIONAL_VERSIONS}" ]; then
-    OLDIFS=$IFS
+# Install additional versions
+if [ -n "$DOTNET_ADDITIONAL_VERSIONS" ]; then
```

Comment:
Does the feature currently behave this way?  I was under the impression that `dotnet --version` will just return the most recent version installed

Reply:
> Does the feature currently behave this way?

Yep

![image](https://user-images.githubusercontent.com/24955913/222280381-120dc771-9220-421e-b634-456ceccc1b6b.png)

Reply:
I have the same impression as Josh. Per the .NET docs, SDK commands always use the latest version unless a `globals.json` file specifies a more specific version.

https://learn.microsoft.com/en-us/dotnet/core/versions/selection#the-sdk-uses-the-latest-installed-version

In the Universal image, I believe `dotnet --version` returns `7` because it's the latest installed version.

Reply:
In other words, I don't think .NET treats one version as "primary" and others as "secondary". I think it just picks the latest version unless there's a `globals.json` file that says otherwise.

Reply:
Currently, additionalVersions are installed with `install_using_dotnet_releases_url` which checks for the OVERRIDE flag and sets .NET root based on it.

https://github.com/devcontainers/features/pull/467/files#diff-3dff5aa292fb61b47ab35a7440d0770b7188768ac421213739525e3aca615cdcL378